### PR TITLE
adding configurable repo version and mirror

### PR DESCRIPTION
**Pull request by zoide**

### CODE REVIEW

File: manifests/params.pp [hunk 1]
```diff
@@ -73,6 +73,7 @@
       $ssl_key                = "${config_dir}//server-key.pem"
       $repo_class             = 'mariadb::repo::debian'
       $wsrep_provider         = '/usr/lib/galera/libgalera_smm.so'
+      $default_mirror = 'http://mirror.aarnet.edu.au/pub/MariaDB'
```
Comment (sorrison) [inline manifests/params.pp]: Please align text

File: manifests/server.pp [hunk 2]
```diff
@@ -52,16 +52,18 @@
...
   $manage_repo             = true,
+  $mirror                  = $mariadb::params::default_mirror,
 ) inherits mariadb::params {
 
   class { 'mariadb':
     package_names  => $client_package_names,
     package_ensure => $client_package_ensure,
     repo_version   => $repo_version,
     manage_repo    => $manage_repo,
+    mirror         => mirror,
```
Comment (sorrison) [inline manifests/server.pp]: Think this should be $mirror

**Comment by zoide** [comment]
All done.

File: manifests/init.pp [hunk 3]
```diff
@@ -27,15 +27,12 @@
 class mariadb (
   $package_ensure = 'present',
   $package_names  = $mariadb::params::client_package_names,
-  $repo_version   = '5.5',
+  $repo_version   = $mariadb::params::repo_version,
```
Comment (sorrison) [inline manifests/init.pp]: this variable doesn't exist, you'll need to add a default to the params class

File: manifests/repo/debian.pp [hunk 4]
```diff
@@ -1,8 +1,11 @@
-class mariadb::repo::debian {
-  $version = $mariadb::repo_version
+class mariadb::repo::debian (
+  $repo_version = $mariadb::params::repo_version,
```
Comment (sorrison) [inline manifests/repo/debian.pp]: whitespace at the end of the line

**Comment by zoide** [comment]
Did all those changes, I hope the Redhat stuff is correct. I don't know very much abut RH.

**Comment by sorrison** [comment]
Fixed with a32f232b985a4a8710736dda3da336acd4fbff3c

**Pull request closed without merging**